zebrad/tests/common/get_block_template_rpcs.rs

```diff
@@ -0,0 +1,164 @@
+
+    use crate::common::{
+        cached_state::{copy_state_directory, start_state_service_with_cache_dir},
+        config::{persistent_test_config, testdir},
+        launch::ZebradTestDirExt,
+        lightwalletd::random_known_rpc_port_config,
+    };
+
+    use super::cached_state::{load_tip_height_from_state_directory, ZEBRA_CACHED_STATE_DIR};
+
+    async fn get_future_block_hex_data(
+        network: Network,
+        zebrad_state_path: &PathBuf,
+    ) -> Result<Option<String>> {
+        tracing::info!(
+            ?zebrad_state_path,
+            "getting cached sync height from ZEBRA_CACHED_STATE_DIR path"
+        );
+
+        let cached_sync_height =
+            load_tip_height_from_state_directory(network, zebrad_state_path.as_ref()).await?;
+
+        let future_block_height = Height(cached_sync_height.0 + 1);
+
+        tracing::info!(
+            ?cached_sync_height,
+            ?future_block_height,
+            "got cached sync height, copying state dir to tempdir"
+        );
+
+        let copied_state_path = copy_state_directory(network, &zebrad_state_path).await?;
+
+        let mut config = persistent_test_config()?;
+        config.state.debug_stop_at_height = Some(future_block_height.0);
+
+        let mut child = copied_state_path
+            .with_config(&mut config)?
+            .spawn_child(args!["start"])?
+            .bypass_test_capture(true);
+
+        while child.is_running() {
+            tokio::task::yield_now().await;
+        }
+
+        let _ = child.kill(true);
+        let copied_state_path = child.dir.take().unwrap();
+
+        let (_read_write_state_service, mut state, _latest_chain_tip, _chain_tip_change) =
+            start_state_service_with_cache_dir(network, copied_state_path.as_ref()).await?;
+        let request = zebra_state::ReadRequest::Block(HashOrHeight::Height(future_block_height));
+
+        let response = state
+            .ready()
+            .and_then(|ready_service| ready_service.call(request))
+            .map_err(|error| eyre!(error))
+            .await?;
+
+        let block_hex_data = match response {
+            zebra_state::ReadResponse::Block(Some(block)) => {
+                hex::encode(block.zcash_serialize_to_vec()?)
+            }
+            zebra_state::ReadResponse::Block(None) => {
+                tracing::info!(
+                "Reached the end of the finalized chain, state is missing block at {future_block_height:?}",
+            );
+                return Ok(None);
+            }
+            _ => unreachable!("Incorrect response from state service: {response:?}"),
+        };
+
+        Ok(Some(block_hex_data))
+    }
+
+    #[allow(clippy::print_stderr)]
+    pub(crate) async fn run() -> Result<(), color_eyre::Report> {
+        let _init_guard = zebra_test::init();
+
+        let mut config = random_known_rpc_port_config(true)?;
+        let network = config.network.network;
+        let rpc_address = config.rpc.listen_addr.unwrap();
```

Comment:
This code seems to be a duplicate of `LightwalletdTestType.zebrad_config()`.

But we'll need to add a `LightwalletdTestType` variant that opens a RPC port, and only requires a cached Zebra state. Like `UpdateZebraCachedStateNoRpc`, but with an RPC port.

Reply:
I'll update this in the PR for #5015